components/builder-api/src/depot/memcached.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use config::Config;
+use hab_core::package::PackageIdent;
+use memcache;
+use std::sync::Arc;
+
+pub struct MemCache {
+    cli: memcache::Client,
+}
+
+impl MemCache {
+    pub fn new(config: Arc<Config>) -> Self {
+        let memcache_host_strings: Vec<String> =
+            config.memcached.iter().map(|h| format!("{}", h)).collect();
+        let memcache_hosts: Vec<&str> = memcache_host_strings.iter().map(AsRef::as_ref).collect();
+        MemCache {
+            cli: memcache::Client::new(memcache_hosts).unwrap(),
+        }
+    }
+
+    pub fn set_package(&mut self, package: PackageIdent, channel: &str) {
+        if package.version.is_none() || package.release.is_none() {
+            debug!("Can't insert non-fully qualified package ident into the cache");
+            return;
+        }
+        match self.cli.set(
+            &format!("{}/{}", package.origin, package.name),
+            package.to_string(),
+            0,
+        ) {
+            Ok(_) => debug!("Saved {:?} to memcached", package.to_string()),
+            Err(_) => error!("Failed to save {:?} to memcached", package.to_string()),
+        };
+        match self.cli.set(
+            &format!("{}/{}/{}", channel, package.origin, package.name),
+            package.to_string(),
+            0,
```

Comment:
Just to be safe, might be good to have an expiration on the sets.. maybe 15 mins?